kdumpctl

```diff
@@ -1668,6 +1672,137 @@ _should_reset_crashkernel() {
+#
+# B) If $KDUMP_STATUS exist, then treat kexec-tools as already installed, and
+#    finding the latest vmcore in the host as specified by kdump.conf,
+#    a) If host is different from $KDUMP_STATUS's record, then update the vmcore
+#       into $KDUMP_STATUS, mark & report this as untested.
+#    b) If host is the same as $KDUMP_STATUS's record,
+#       1) If the timestamp of the vmcore is equal or smaller(older) than
+#          $KDUMP_STATUS's record, then only report the status of $KDUMP_STATUS's
+#          record.
+#       2) If the timestamp of the vmcore is larger(newer) than $KDUMP_STATUS's
+#          record, then update the vmcore into $KDUMP_STATUS, mark & report this
+#          as tested.
+#
+# host is a string like "/var/crash" for local dump, "root@1.2.3.4:/var/crash"
+# for ssh dump, "1.2.3.4:/var/crash" for nfs dump.
+check_kdump_tested()
+{
+	local _record_of_current _current_core_host _current_core_file
+	local _current_core_timestamp _current_core_date
+	local _record_of_kdstatus _kdstatus_core_host _kdstatus_core_file
+	local _kdstatus_core_timestamp _kdstatus_test_status
+	local _ret
+
+	_record_of_current=$(get_latest_dump)
+	if [[ $? -ne 0 ]]; then
+		dwarn "Test fail: Due to previous error."
+		return
+	fi
+	# There may be no vmcores find by get_latest_dump
+	if [[ $(echo $_record_of_current | tr ' ' '\n' | wc -l) -lt 3 ]]; then
+		_record_of_current="dummy_host dummy_vmcore 0"
+	fi
+	_current_core_host=$(echo $_record_of_current | awk '{print $1}')
+	_current_core_file=$(echo $_record_of_current | awk '{print $2}')
+	_current_core_timestamp=$(echo $_record_of_current | awk '{print $3}')
+	_current_core_date=$(echo "@"$_current_core_timestamp | xargs date -d 2> /dev/null)
+
+	if [[ ! $_current_core_file =~ .*vmcore(\.flat)?$ ]]; then
+		dwarn "Test fail: Latest vmcore saving is incomplete."
+		return
+	fi
+
+	if [[ ! -s $KDUMP_STATUS ]]; then
+		echo "$_record_of_current untested" > $KDUMP_STATUS
+		dwarn "Test fail: NOT tested yet"
+		return
+	fi
+
+	_record_of_kdstatus=$(cat $KDUMP_STATUS)
+	_kdstatus_core_host=$(echo $_record_of_kdstatus | awk '{print $1}')
+	_kdstatus_core_file=$(echo $_record_of_kdstatus | awk '{print $2}')
+	_kdstatus_core_timestamp=$(echo $_record_of_kdstatus | awk '{print $3}')
+	_kdstatus_core_date=$(echo "@"$_kdstatus_core_timestamp | xargs date -d 2> /dev/null)
+	_kdstatus_test_status=$(echo $_record_of_kdstatus | awk '{print $4}')
+
+	if [[ $_kdstatus_core_host == "dummy_host" ]]; then
+		if [[ $_current_core_host == "dummy_host" ]]; then
+			dwarn "Test fail: NOT tested yet"
+		else
+			echo "$_record_of_current tested" > $KDUMP_STATUS
+			dinfo "Test success: Last tested on $_current_core_date"
+		fi
+		return
+	fi
+
+	if [[ $_kdstatus_core_host == $_current_core_host ]]; then
+		if [[ $_current_core_timestamp -gt $_kdstatus_core_timestamp ]]; then
+			echo "$_record_of_current tested" > $KDUMP_STATUS
+			dinfo "Test success: Last tested on $_current_core_date"
```

Comment:
maybe use "Tesed: last tested on ..."   "success" should be fore vmcore saved case, and fail should be for no vmcore or incomplete?